src/hats_import/catalog/arguments.py

```diff
@@ -134,9 +138,53 @@ def to_table_properties(
             "hats_max_rows": self.pixel_threshold,
             "hats_order": highest_order,
             "moc_sky_fraction": f"{moc_sky_fraction:0.5f}",
-        } | self.extra_property_dict()
+        }
         return TableProperties(**info)
 
+    @classmethod
+    def reimport_from_hats(
+        cls, path: str | Path | UPath, output_dir: str | Path | UPath, **kwargs
+    ) -> ImportArguments:
+        """Generate the import arguments to reimport a HATS catalog with different parameters
+
+        Args:
+            path (str | Path | UPath): the path to the existing HATS catalog to reimport
+            output_dir (str | Path | UPath): the path to output the reimported catalog to
+            kwargs: any import arguments to update from the existing catalog. Can be any argument usually
+                passed to :func:`ImportArguments`
+
+        Returns:
+            A ImportArguments object with the arguments from the existing catalog, and any updates from kwargs
+        """
+
+        path = get_upath(path)
+
+        catalog = hats.read_hats(path)
+
+        column_names = catalog.schema.names if catalog.schema is not None else None
+
+        in_file_paths = glob.glob(
```

Comment:
Paths should still be instances of UPath. You may want to go ahead and add a test case for this inside hats-cloudtests, as I suspect this may cause problems if reading from non-local file systems.